### DISCUSSION TITLE
High memory usage in SpudSoft Xlsx Emitter

### DISCUSSION THREAD

**Discussion by redkark** [eclipse-birt/birt#1318] [Q&A] [answered]
I observe high memory usage when using Spudsoft Emitter while generate XLSX output for larger data volumes. I am using the BIRT runtime v4.12 in my Java code to generate the reports. I looked at certain posts where it was suggested to use the ExcelEmitter.Extract in the render options to instruct the spudsoft emitter to use the "streaming" SXSSFWorkbook from Apache POI. However it did not have desired effect. Upon looking at the code for XslxEmitter, I noticed it uses the XSSFWorkbook with no option provided to switch to the SXSSFWorkbook. Would like to know if it is possible to configure spudsoft emitter to make use of the SXSSSFWorkbook, via the render options or other mechanisms?

**Comment by speckyspooky** [comment]
Currently there isn't any option implemented to switch from XSSFWorkbook to SXSSSFWorkbook.
This would be an idea to add a new ExcelEmitter-option to define the excel-workbook-mode.
But it must be implemented like a new feature.

**Comment by speckyspooky** [comment, marked as answer]
With the latest BIRT-snapshop and with the upcoming version 4.14 a new spudsoft-emitter option will be available.
Based on your note I added the new user property "ExcelEmitter.StreamingXlsx".
You can switch on the SXSSF-stream-handling of xlsx-files.

"true": SXSSF-Workbook is active
"false" (default): XSSF-Workbook is used, the current used handling

Be aware the internal handling isn't the same of XSSF & SXSSF and it could have impacts on the result.

**Reply by redkark** [reply]
Many thanks @speckyspooky for you prompt support on this 👍
